.github/workflows/main.yml

```diff
@@ -39,13 +54,7 @@ jobs:
       - uses: mamba-org/setup-micromamba@v1
         with:
           # TODO: Use a conda environment file used for the diracx/base container image
-          environment-name: test-env
-          create-args: >-
-            python=3.11
-            m2crypto
-            python-gfal2
-            mypy
-            pip
+          environment-file: environment.yml
```

Comment:
it's not this `environment.yml` but the one from `container-images` that should be used

Reply:
This `environment.yml` defines the environment where all of us are working locally, right? So, it's the natural environment where all unit tests run. 

In `container-images` there are many `environment.yml` files. What is here is, permit me that, the superset of all of them.

Reply:
absolutely, but this is what we want to try to avoid. We want to have the minimal set of dependencies installed to keep control of what we *actually* need, and this is why we did not blindly put the `environment.yml` before

Reply:
So, you want to repeat, for ONLY the unit tests run by the CI, the ~same environment (because it is not going to be very different) that we all have in our local machines, and that we all use for running the unit tests locally? Meaning that I can run the unit tests locally, they all pass, then create the PR only to discover that there they are failing. If there's a version to restrict (like for the one for typer), that, like it or not, we have to do it from time to time, then we would need to apply that fix in even one more place.

There is already a [noted duplication between setup.cfg and environment.yml](https://github.com/DIRACGrid/diracx/blob/main/docs/VERSIONING.md#dependencies) and I do not think we should have even more.

The environment where the unit tests run is already NOT what it is run in production. Which is fair and fine. But that is why the integrations tests are there.

Reply:
We expect the CI to run in an environment which is different to out local machines as we intetionally only install the bare minimum in the CI. We also let it use the newest possible version to try and catch issues before they affect the harder/slower to debug tests.

Maybe we want to drop the environment.yaml file entirely but that's a separate discussion to start and maybe we want to use [something else to harmonise this](https://pixi.sh/latest/features/multi_environment/) but we've not got around to trying it.